metadata-ingestion/src/datahub/ingestion/source/ge_data_profiler.py

```diff
@@ -793,6 +793,13 @@ def update_dataset_batch_use_sampling(self, profile: DatasetProfileClass) -> Non
                 self.dataset._table = sa.text(temp_table_name)
                 logger.debug(f"Setting table name to be {self.dataset._table}")
 
+                # We can alternatively use `self._get_dataset_rows(profile)` to get
+                # exact count of rows in sample, as actual rows involved in sample
+                # may be slightly different (more or less) than configured `sample_size`.
+                # However not doing so to start with, as that adds another query overhead
+                # plus approximate metrics should work for sampling based profiling.
+                profile.rowCount = self.config.sample_size
```

Comment:
Imo the row count should still ultimately reflect the full number of rows, and the size of the sample should be stored in the partition spec

Reply:
The actual table's rowCount here is always overridden in bigquery source from information_schema, whenever partitionSpec is set. partitionSpec is set whenever limit or sampling is used in profiling or only latest partition is used for partitioned tables. The computed rowCount in ge_data_profiler.py is used primarily for column level profiling computation only. (this is also the case for snowflake and redshift)

Reply:
I'm open to handling this only for null counts computation. Attempted changing profile aspect to make sure that we don't run  into this for any other calculation  - maybe in future.

Reply:
Yup that might result in more understandable code

At minimum, we need a comment here explaining why it's ok to override, but ideally we just tweak the null count / unique percentage calculations